src/Graphics/Texture.cs

```diff
@@ -313,8 +323,80 @@ SurfaceFormat format
 					case FOURCC_DXT5:
 						format = SurfaceFormat.Dxt5;
 						break;
-					case FOURCC_BPTC:
-						format = SurfaceFormat.Bc7EXT;
+					case FOURCC_DX10:
+						// If the fourCC is DX10, there is an extra header with additional format information.
+						uint dxgiFormat = reader.ReadUInt32();
+
+						// These values are taken from the DXGI_FORMAT enum.
+						switch (dxgiFormat)
+						{
+							case 2:
+								format = SurfaceFormat.Vector4;
+								break;
+
+							case 10:
+								format = SurfaceFormat.HalfVector4;
+								break;
+
+							case 71:
+								format = SurfaceFormat.Dxt1;
+								break;
+
+							case 74:
+								format = SurfaceFormat.Dxt3;
+								break;
+
+							case 77:
+								format = SurfaceFormat.Dxt5;
+								break;
+
+							case 98:
+								format = SurfaceFormat.Bc7EXT;
+								break;
+
+							case 99:
+								format = SurfaceFormat.Bc7SrgbEXT;
+								break;
+
+							default:
+								throw new NotSupportedException(
+									"Unsupported DDS texture format"
+								);
+						}
+
+						uint resourceDimension = reader.ReadUInt32();
+
+						// These values are taken from the D3D10_RESOURCE_DIMENSION enum.
+						switch (resourceDimension)
+						{
+							case 0: // Unknown
+							case 1: // Buffer
+								throw new NotSupportedException(
+									"Unsupported DDS texture format"
+								);
```

Comment:
Worth having an empty default handler, genuinely thought this would throw 100% until I stared at it for a minute.

Reply:
Done